src/context/viewer-context.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { Vault } from "@hyperion-framework/vault";
+
+const ViewerStateContext = React.createContext();
+const ViewerDispatchContext = React.createContext();
+
+const defaultState = {
+  activeCanvas: null,
+  vault: new Vault(),
+  isLoaded: false,
+};
+
+function viewerReducer(state, action) {
+  switch (action.type) {
+    case "updateActiveCanvas": {
+      return {
+        ...state,
+        activeCanvas: action.canvasId,
+      };
+    }
+    case "updateIsLoaded": {
+      return {
+        ...state,
+        isLoaded: { ...action.isLoaded },
+      };
+    }
+    default: {
+      throw new Error(`Unhandled action type: ${action.type}`);
+    }
+  }
+}
+
+interface ViewerProviderProps {
+  initialState: object;
+  manifestId: string;
+}
+
+function ViewerProvider<ViewerProviderProps>({
+  initialState = defaultState,
+  manifestId,
+  children,
+}) {
+  const [state, dispatch] = React.useReducer(viewerReducer, initialState);
+
+  return (
+    <ViewerStateContext.Provider
+      value={{
+        manifestId,
+        activeCanvas: state.activeCanvas,
+        vault: state.vault,
+        isLoaded: state.isLoaded,
+      }}
```

Comment:
@mathewjordan Since we mostly pushing `state.someProperty` into the Context's `value`, can we simplify this and just do:

`<ViewerStateContext.Provider value={state} ..../>` ?     Do we need `manifestId` in the Context values or is that just a holdover from earlier?